### ISSUE TITLE
[Feature Request] Add killswitch to disable based on env

### ISSUE THREAD

**Issue by Stanzilla** [ultrabug/mkdocs-static-i18n#32]
mkdocs 1.2 is out now and has support for `ENV` checks in configs, this would allow me to get rid of having to use two different `mkdocs.yml` but `mkdocs-static-i18n` does not have an easy way to disable it.

Would be nice if there was an `enabled: true|false` key I could set.

**Comment by ultrabug** [comment]
> but mkdocs-static-i18n does not have an easy way to disable it

I'm sorry but disable what exactly? I don't understand what that switch you're talking about is supposed to do?

Can you give me more details? I'll happily help as usual :)

**Comment by Stanzilla** [comment]
The whole plugin basically.

My goal here is to only run the multi-site generation on the production and test environments but not during local testing since it increases the build times a lot. 

```yml
  - i18n:
  	  enabled: !ENV MKDOCS_I18N
      default_language: en
      languages:
        fr: Français (Beta)
```

As an example, I could now set the `MKDOCS_I18N` env variable to true/false.

**Comment by Stanzilla** [comment]
Or maybe even switches for every language?

```yml
  - i18n:
      default_language: en
      languages:
        fr: !ENV MKDOCS_I18N_FR
		es: !ENV MKDOCS_I18N_ES
		tw: !ENV MKDOCS_I18N_TW
```

so if a language key has `false` at a value, it does not get considered during compilation.

**Comment by ultrabug** [comment]
> The whole plugin basically.

oh! I see.

It looks to me that your use case was one of the things behind https://github.com/mkdocs/mkdocs/issues/2218

Did you consider using configuration inheritance coupled with ENV vars to achieve what you're looking for?

**Comment by Stanzilla** [comment]
@ultrabug yeah I wanted to avoid using inheritance and just use the ENVs. But I can try it if you don't think the switches are in scope for your plugin!

**Comment by ultrabug** [comment]
> so if a language key has false at a value, it does not get considered during compilation.

I kind of dislike options with different meanings / behaviors so if we were to move on this, I'd make a separate option tbh

**Comment by Stanzilla** [comment]
> > so if a language key has false at a value, it does not get considered during compilation.
> 
> I kind of dislike options with different meanings / behaviors so if we were to move on this, I'd make a separate option tbh

So the global disable?

**Comment by ultrabug** [comment]
Yeah on principle, but I'd like to find another way tbh, I don't like the idea of plugins having to have this kind of option and recent 1.2 features look to provide that possibility

**Comment by Stanzilla** [comment]
I wanted to avoid having two config files and with the 1.2 feature it looks like I would still have to have two? Also it needs the disable option as well? I might have read https://github.com/mkdocs/mkdocs/issues/2218#issuecomment-743312764 wrong though.

**Comment by ultrabug** [comment]
I just tried and could achieve what I believe you're looking for after [reading this comment](https://github.com/mkdocs/mkdocs/issues/2218#issuecomment-843488261)

It works good by adding a file to define plugins per environment, you still have only one `mkdocs.yml` file so you do not need to adjust your automation.

in `mkdocs.yml` you replace the whole `plugins:` definition by an INHERIT:

```
INHERIT: !ENV PLUGINS
```

then create `mkdocs-production.yml` and put your full plugins definition inside like this for example:

```
plugins:
  - search
  - i18n:
      default_language: en
      languages:
        en: english
        fr: français
      nav_translations:
        fr:
          Topic1: Sujet1
          Topic2: Sujet2
```

then create `mkdocs-development.yml` and put your development plugins inside, like this for exemple:

```
plugins:
  - search
```

now you can build your development with the minimal set of plugins you want version using

```
PLUGINS=mkdocs-development.yml mkdocs build
```

and the production using 

```
PLUGINS=mkdocs-production.yml mkdocs build
```

Would that sound reasonable to you?

**Comment by Stanzilla** [comment]
Hrm no, not at all because that is one file more than I have now. Currently I just have two configs that I use based on the environment and call with `--config-file`

**Comment by ultrabug** [comment]
But you have to duplicate the whole thing with your current implementation whereas my proposal allows you to focus on the specifics of a given environment. And most importantly here, you'd only have ONE mkdocs.yml to work with.

That being said, for the sake of simplicity and because you're the first stargazer of this projet ( :wink: ) I will implement a `enabled` option which defaults to true and that you can override with ENV.

Stay tuned

**Comment by Stanzilla** [comment]
I mean yeah that is indeed true. And aww okay, thank you!

**Comment by ultrabug** [comment]
@Stanzilla I still think that the plugin should sort the files and fix the navs to avoid warnings all over the place when disabled? that way only the actual build of the languages would be disabled but not the rest.

This means that you would only build the default language of the site, not the rest.

or do you expect the whole plugin to be skipped?

**Comment by Stanzilla** [comment]
from a UX standpoint, I think disabled should mean completely disabled...but you do have a point, hrm.

**Comment by ultrabug** [comment]
^ this would also allow developers to change the `default_language` as they wish (even with ENV!) and only (re)build for that

- **default_language_only** (default: false): allows to only build the default language, skipping the configured languages (useful for development when rebuilding all the languages is slow)

something like this

**Comment by Stanzilla** [comment]
oh that does sound nice indeed!

**Comment by ultrabug** [comment]
plz check #38

**Comment by ultrabug** [comment]
available in version 0.16 mate, enjoy :star:
